runtime/vm/createramclass.cpp

```diff
@@ -2894,12 +2894,19 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 				} else {
 					arity = 1;
 					leafComponentType = elementClass;
+					/* For arrays of valueType elements (where componentType is a valuetype), the arrays themselves are not
+					 * valuetypes but they should inherit the layout characteristics (ie. flattenable, etc.)
+					 * of the valuetype elements. A 2D (or more) array of valuetype elements (where leafComponentType is a Valuetype but
+					 * componentType is not) is not direct array array of valuetype elements, therefore it should not inherit any layout
+					 * properties from the leafComponentType. A 2D array is an array of references so it can never be flattened, however, its
+					 * elements may be flattened arrays.
+					 */
+					ramArrayClass->classFlags |= (J9ClassIsFlattened | J9ClassLargestAlignmentConstraintReference | J9ClassLargestAlignmentConstraintDouble) & elementClass->classFlags;
```

Comment:
Please invert the `&` operands - this is why I was confused thinking you were adding flags.